src/test/java/org/nuxeo/wopi/lock/TestLock.java

```diff
@@ -0,0 +1,190 @@
+@RunWith(FeaturesRunner.class)
+@Features(PlatformFeature.class)
+@Deploy("org.nuxeo.ecm.platform.web.common")
+@Deploy("org.nuxeo.wopi.rest.api")
+public class TestLock {
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected EventProducer eventProducer;
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    protected DocumentModel doc;
+
+    protected DocumentModel expiredLockDoc;
+
+    protected String fileId;
+
+    protected String expiredLockFileId;
+
+    @Before
+    public void setUp() {
+        doc = session.createDocumentModel("/", "doc", "File");
+        doc = session.createDocument(doc);
+        fileId = FileInfo.computeFileId(doc, FILE_CONTENT_PROPERTY);
+
+        expiredLockDoc = session.createDocumentModel("/", "expiredLockDoc", "File");
+        expiredLockDoc = session.createDocument(expiredLockDoc);
+        expiredLockFileId = FileInfo.computeFileId(expiredLockDoc, FILE_CONTENT_PROPERTY);
+    }
+
+    @Test
+    public void testLockDirectory() throws InterruptedException {
+        // get unknown lock
+        assertNull(LockHelper.getLock("unknownFileId"));
+
+        // add lock / get lock
+        LockHelper.addLock(fileId, "wopiLock");
+        assertEquals("wopiLock", LockHelper.getLock(fileId));
+
+        // update lock
+        LockHelper.updateLock(fileId, "updatedWopiLock");
+        assertEquals("updatedWopiLock", LockHelper.getLock(fileId));
+
+        // refresh lock
+        try (Session session = LockHelper.openLockDirectorySession()) {
+            long timestamp = (long) session.getEntry(fileId).getProperty(LOCK_DIRECTORY_SCHEMA_NAME,
+                    LOCK_DIRECTORY_TIMESTAMP);
+            Thread.sleep(10);
+            LockHelper.refreshLock(fileId);
+            long refreshedTimestamp = (long) session.getEntry(fileId).getProperty(LOCK_DIRECTORY_SCHEMA_NAME,
+                    LOCK_DIRECTORY_TIMESTAMP);
+            assertTrue(Instant.ofEpochMilli(refreshedTimestamp).isAfter(Instant.ofEpochMilli(timestamp)));
+        }
+
+        // remove lock
+        LockHelper.removeLock(fileId);
+        assertNull(LockHelper.getLock(fileId));
+
+        // get expired locks
+        LockHelper.addLock(fileId, "wopiLock");
+        assertEquals("wopiLock", LockHelper.getLock(fileId));
+
+        LockHelper.addLock(expiredLockFileId, "expiredWopiLock");
+        assertEquals("expiredWopiLock", LockHelper.getLock(expiredLockFileId));
+
+        Map<String, List<DocumentModel>> expiredLocksByRepository = LockHelper.getExpiredLocksByRepository();
+        assertEquals(1, expiredLocksByRepository.size());
+        String repositoryName = session.getRepositoryName();
+        List<DocumentModel> expiredLocks = expiredLocksByRepository.get(repositoryName);
+        assertNotNull(expiredLocks);
+        assertTrue(expiredLocks.isEmpty());
+
+        // force expiration for expiredLockFileId
+        try (Session session = LockHelper.openLockDirectorySession()) {
```

Comment:
Maybe put this in a common method as this is duplicated

Reply:
OK, in fact I refactored the LockHelper